### PR TITLE
Suggest command based on aliases

### PR DESCRIPTION
**Pull request by sigurdm**
Also introduce a concept of suggestionAliases (alternative names that cannot be used as a command name, but they will be used for suggesting the actual command).

The motivation comes from this suggestion: https://github.com/dart-lang/sdk/issues/50971#issuecomment-1412495274

That would allow us to give a good suggestion if someone writes `dart packages get` -> `Did you mean pub?`.

### CODE REVIEW

File: lib/command_runner.dart [hunk 1]
```diff
@@ -412,6 +419,17 @@ abstract class Command<T> {
   /// This is intended to be overridden.
   List<String> get aliases => const [];
 
+  /// Alternate non-functional names for this command.
+  ///
+  /// These names won't be used in the documentation, and also they will work
```
Comment (jakemac53) [inline lib/command_runner.dart:424]:
```suggestion
  /// These names won't be used in the documentation, and also they won't work
```

**Review by jakemac53 — COMMENTED** [review]
This LGTM but we do need a pubspec/changelog update.

I also can't remember for certain whether it is supported to implement `Command` but I don't think so? It would be breaking if so, but I am fairly certain we always extend?

**Comment by sigurdm** [comment]
>  It would be breaking if so, but I am fairly certain we always extend?

Good point about breakiness. I searched internally for usages, and the only place we `implement` this is in dart2js_info where it is used for a mixin. So in practice this seems quite non-breaking. However I don't think it is documented anywhere?

It seems we have precedence of adding properties to `Command` without a breaking release: https://github.com/dart-lang/args/pull/202.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## 2.4.0-dev
```
Comment (jakemac53) [inline CHANGELOG.md:1]: Want to go right to a stable release? We don't have any other changes I know of coming.

**Review by jakemac53 — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (sigurdm) [reply CHANGELOG.md:1]: Ok with me.

**Comment by jakemac53** [comment]
Filed https://github.com/dart-lang/ecosystem/issues/56 not sure what is up with that rpc error

**Comment by devoncarew** [comment]
hmm, this:

```
RpcException: Forbidden
#0      Github.callRestApiPost.<anonymous closure> (package:firehose/src/github.dart:66:13)
<asynchronous suspension>
#1      Github.createComment (package:firehose/src/github.dart:104:18)
...
```

was - after a successful package validation - we tried to write a comment back to the PR saying something like `tag with xxx to invoke a publish`.

**Comment by devoncarew** [comment]
Hmm, as a mea culpa, I published this without first making sure that it had rolled through the sdk and google3. I'll watch for any issues downstream.

**Pull request merged**